pkg/repositories/gormimpl/dataset.go

```diff
@@ -44,7 +44,7 @@ func (h *dataSetRepo) Get(ctx context.Context, in models.DatasetKey) (models.Dat
 	defer timer.Stop()
 
 	var ds models.Dataset
-	result := h.db.Where(&models.Dataset{DatasetKey: in}).First(&ds)
+	result := h.db.Preload("PartitionKeys").First(&ds, &models.Dataset{DatasetKey: in})
```

Comment:
what is the impact of Preload here? Docs on it are kinda confusing.

Reply:
Preload basically tacks on a call to get the partition keys for the Dataset returned from `.First(&ds, &models.Dataset{DatasetKey: in})`. This line results in two SQL calls if the Dataset is found:
```
SELECT * from datasets WHERE dataset.project=... AND ...
SELECT * from partition_keys where partition_key.dataset_uuid='<uuid of result from first select>'
```

If the Dataset is not found only the first sql statement will be issued.